### DISCUSSION TITLE
Draw part of image on another image

### DISCUSSION THREAD

**Discussion by TheBoneJarmer** [SixLabors/ImageSharp#2102] [Ideas]
Hey guys

Using  _System.Drawing.BitMap_ and _System.Drawing.Graphics_ it was possible to render a part of an image, let's say a tileset for example, on another using a source rect and dest rect. Microsoft dropped Linux support for the NuGet package _System.Drawing_ so I went for ImageSharp as it seemed like a good alternative yet this feature seems to be either missing or well hidden as I cannot seem to find it.

With kind regards,
TheBoneJarmer

**Comment by ynse01** [comment]
If I understand correctly, you want to clone a part of a source image into a destination image.
Like so:
`Image dest = source.Clone(c => c.Crop(tileRectangle));`

Where `tileRectangle` is the area of the source image you want to clone.

**Reply by br3aker** [reply]
It can be solved this way but it'll take extra resources. We already have a draw image processor with `Point position` parameter, seems good to implement `DrawImage(Image, Point, Rectangle)` overload only if it's not implemented in `ImageSharp.Drawing`.

**Reply by JimBobSquarePants** [reply]
Yeah no reason I can think of we wouldn’t be able to select a source rectangle

**Reply by sunkin351** [reply]
> It can be solved this way but it'll take extra resources. We already have a draw image processor with `Point position` parameter, seems good to implement `DrawImage(Image, Point, Rectangle)` overload only if it's not implemented in `ImageSharp.Drawing`.

I have need of this, and would love to see it implemented.

**Reply by dvisentin-freelance** [reply]
> I have need of this, and would love to see it implemented.

Me too.

**Reply by JimBobSquarePants** [reply]
[Here](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Processing.DrawImageExtensions.html#SixLabors_ImageSharp_Processing_DrawImageExtensions_DrawImage_SixLabors_ImageSharp_Processing_IImageProcessingContext_SixLabors_ImageSharp_Image_SixLabors_ImageSharp_Point_SixLabors_ImageSharp_Rectangle_SixLabors_ImageSharp_GraphicsOptions_)
